pkg/runtime/codec.go

```diff
@@ -22,6 +22,24 @@ import (
 	"k8s.io/kubernetes/pkg/util/yaml"
 )
 
+// Encode is a convenience wrapper for encoding to a []byte from an Encoder
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func Encode(e Encoder, obj Object) ([]byte, error) {
+	return e.Encode(obj)
+}
+
+// Decode is a convenience wrapper for decoding data into an Object.
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func Decode(d Decoder, data []byte) (Object, error) {
+	return d.Decode(data)
+}
+
+// DecodeInto performs a Decode into the provided object.
+// TODO: these are transitional interfaces to reduce refactor cost as Codec is altered.
+func DecodeInto(d Decoder, data []byte, into Object) error {
+	return d.DecodeInto(data, into)
```

Comment:
Will the interface contain DecodeInto()?
Will DecodeIntoWithSpecifiedVersionKind be implemented using d.Decode()?